File: rb/features.rb
# MeteoprogWeather SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module MeteoprogWeatherFeatures
  def self.make_feature(name)
    case name
    when "base"
      MeteoprogWeatherBaseFeature.new
    when "test"
      MeteoprogWeatherTestFeature.new
    else
      MeteoprogWeatherBaseFeature.new
    end
  end
end

File: py/features.py
# MeteoprogWeather SDK feature factory

from feature.base_feature import MeteoprogWeatherBaseFeature
from feature.test_feature import MeteoprogWeatherTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MeteoprogWeatherBaseFeature(),
        "test": lambda: MeteoprogWeatherTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
